homeassistant/components/sensor/aftership.py

```diff
@@ -0,0 +1,130 @@
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pyaftership==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Information provided by AfterShip'
+
+DEFAULT_NAME = 'aftership'
+
+CONF_TITLE = 'title'
+CONF_SLUG = 'slug'
+
+CONF_TRACKING_NUMBER = 'tracking_number'
+
+ICON = 'mdi:package-variant-closed'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    })
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the AfterShip sensor platform."""
+    from pyaftership.tracker import Tracking
+
+    apikey = config[CONF_API_KEY]
+    name = config[CONF_NAME]
+
+    session = async_get_clientsession(hass)
+    aftership = Tracking(hass.loop, session, apikey)
+
+    await aftership.get_trackings()
+
+    if not aftership.meta or aftership.meta['code'] != 200:
+        _LOGGER.error("No tracking data found. Check API key is correct: %s",
+                      aftership.meta)
+        return
+
+    sensor = [AfterShipSensor(aftership, name)]
+    async_add_entities(sensor, True)
+
+
+class AfterShipSensor(Entity):
+    """Representation of a AfterShip sensor."""
+
+    def __init__(self, aftership, name):
+        """Initialize the sensor."""
+        self._attributes = {}
+        self._name = name
+        self._state = None
+        self.aftership = aftership
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return 'packages'
+
+    @property
+    def device_state_attributes(self):
+        """Return attributes for the sensor."""
+        return self._attributes
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return ICON
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self):
+        """Get the latest data from the AfterShip API."""
+        await self.aftership.get_trackings()
+
+        if not self.aftership.meta:
+            _LOGGER.error("Unkown errors when querying. See logs for details.")
+            return
+        if self.aftership.meta['code'] != 200:
+            _LOGGER.error("Errors when querying AfterShip. %s",
+                          str(self.aftership.meta))
+            return
+
+        status_to_ignore = {'Delivered'}
+        status_counts = {}
+        not_delivered_count = 0
+
+        for tracking in self.aftership.trackings['trackings']:
+            status = tracking['tag']
+            name = tracking['tracking_number']
+            status_counts[status] = status_counts.get(status, 0)+1
```

Comment:
Is the status key lowercase snakecase? That's our standard for state attribute keys.

Reply:
the values are camel case with spaces in aftership api, see list at https://help.aftership.com/hc/en-us/articles/360007823253

want to lower case it here when storing in state ?

Reply:
That would be good!

Reply:
Please update this and I think this PR is ready to merge.

Reply:
sorry missed the ping - pushed an update that fixes this. it just stores lower values now.